sample-chat/src/main/java/com/quickblox/sample/chat/managers/DialogsManager.java

```diff
@@ -98,8 +102,19 @@ public void onSystemMessageReceived(QBChatMessage systemMessage){
         if (isMessageCreatingDialog(systemMessage)) {
             QBChatDialog chatDialog = buildChatDialogFromSystemMessage(systemMessage);
             chatDialog.initForChat(QBChatService.getInstance());
+
+            try {
+                DiscussionHistory history = new DiscussionHistory();
+                history.setMaxStanzas(0);
+                chatDialog.join(history);
```

Comment:
onSystemMessageReceived is called on main thread, so you should not use sync join method here, please use async version instead